nat-lab/tests/utils/analytics/event_validator.py

```diff
@@ -341,66 +347,94 @@ def add_validator_list(self, validators: List[Validator]) -> Self:
         return self
 
     def add_rtt_validators(
-        self, nodes_ip_stack: Dict[str, Optional[IPStack | None]]
+        self,
+        ip_stacks: List[Optional[IPStack]],
     ) -> Self:
-        primary_node_ip_stack = nodes_ip_stack.pop(self._node_fingerprint)
-        for fingerprint in nodes_ip_stack:
-            secondary_node_ip_stack = nodes_ip_stack.get(fingerprint)
-            if secondary_node_ip_stack is not None:
+        node_ip_stacks = dict(zip(NODES_FINGERPRINTS, ip_stacks))
+        primary_node_ip_stack = node_ip_stacks.pop(self.get_node_fingerprint())
+
+        for fingerprint in node_ip_stacks:
+            secondary_node_ip_stack = node_ip_stacks.get(fingerprint)
+
+            if (
+                secondary_node_ip_stack is not None
+                and primary_node_ip_stack is not None
+            ):
+                (
+                    rtt_c,
+                    rtt_dnc,
+                    rtt_loss_c,
+                    rtt_loss_dnc,
+                    rtt6_c,
+                    rtt6_dnc,
+                    rtt6_loss_c,
+                    rtt6_loss_dnc,
+                ) = (None, None, None, None, None, None, None, None)
+                rtt_eq, rtt_loss_eq, rtt6_eq, rtt6_loss_eq = "", "", "", ""
+
+                match connectivity_stack(
```

Comment:
The match here too will need to be an if-statement

Reply:
Fixed